mmv1/third_party/terraform/services/container/resource_container_cluster.go.tmpl

```diff
@@ -1722,13 +1728,120 @@ func ResourceContainerCluster() *schema.Resource {
+										Type:             schema.TypeBool,
+										Optional:         true,
+										DiffSuppressFunc: suppressWhenIpEndpointsDisabled,
+										ConflictsWith:    []string{"private_cluster_config.0.master_global_access_config.0.enabled"},
+										Description:      `Controls whether the control plane's private endpoint is accessible from sources in other regions.`,
+									},
+									"authorized_networks_config": {
+										Type:             schema.TypeList,
+										Optional:         true,
+										DiffSuppressFunc: suppressWhenIpEndpointsDisabled,
+										ConflictsWith:    []string{"master_authorized_networks_config"},
+										MaxItems:         1,
+										Description:      `Configuration of authorized networks. If enabled, restricts access to the control plane based on source IP.`,
+										Elem:             masterAuthorizedNetworksConfig,
+									},
+									"public_endpoint": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Computed:    true,
+										Description: `The external IP address of this cluster's control plane. Only populated if enabled.`,
+									},
+									"private_endpoint": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Computed:    true,
+										Description: `The internal IP address of this cluster's control plane. Only populated if enabled.`,
+									},
+									"private_endpoint_subnetwork": {
+										Type:             schema.TypeString,
+										Optional:         true,
+										DiffSuppressFunc: suppressWhenIpEndpointsDisabled,
+										ConflictsWith:    []string{"private_cluster_config.0.private_endpoint_subnetwork"},
+										Description:      `Subnet to provision the master's private endpoint during cluster. Specified in projects/*/regions/*/subnetworks/* format.`,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"network_config": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Description: `Configuration for cluster networking.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"default_enable_private_nodes": {
+							Type:          schema.TypeBool,
+							Optional:      true,
+							ConflictsWith: []string{"private_cluster_config.0.enable_private_nodes"},
+							Description:   `Controls whether by default nodes have private IP addresses only.`,
+						},
+					},
+				},
+			},
+
 			"private_cluster_config": {
 				Type:             schema.TypeList,
 				MaxItems:         1,
 				Optional:         true,
 				Computed:         true,
 				DiffSuppressFunc: containerClusterPrivateClusterConfigSuppress,
-				Description:      `Configuration for private clusters, clusters with private nodes.`,
+				Description:      `Configuration for private clusters, clusters with private nodes. Beware that most of the fields have been deprecated in favor of control_plane_endpoints_config.`,
```

Comment:
for here + docs, "It's recommended that new configurations use control_plane_endpoints_config for any settings that are available there"?

Reply:
That sounds better.